chain/chain/src/store.rs

```diff
@@ -1939,6 +2054,9 @@ impl<'a> ChainStoreUpdate<'a> {
         if let Some(t) = self.head.take() {
             store_update.set_ser(ColBlockMisc, HEAD_KEY, &t).map_err::<Error, _>(|e| e.into())?;
         }
+        if let Some(t) = self.tail.take() {
+            store_update.set_ser(ColBlockMisc, TAIL_KEY, &t).map_err::<Error, _>(|e| e.into())?;
```

Comment:
It seems that you just followed the style, but `map_err(|e| e.into())` is useless since `?` operator does that, so you can just remove that completely (update the rest of the code as well):

```rust
store_update.set_ser(ColBlockMisc, TAIL_KEY, &t)?`
```

Reply:
Updated once, don't want to update all in this PR, unrelated too much